app/src/main/kotlin/com/flxrs/dankchat/OAuthResult.kt

```diff
@@ -0,0 +1,8 @@
+package com.flxrs.dankchat
+
+sealed class OAuthResult {
+    // This is named UserName instead of ValidUser because it's also used in case of fallback
+    data class UserName(val username: String) : OAuthResult()
+    object OAuthTokenInvalid : OAuthResult()
+    object OAuthValidationFailure : OAuthResult()
+}
```

Comment:
Some naming updates I would like:
```suggestion
sealed class ValidationResult {
    data class User(val username: String) : ValidationResult()
    object TokenInvalid : ValidationResult()
    object Failure : ValidationResult()
}
```